modules/mono/mono_gd/gd_mono_assembly.cpp

```diff
@@ -209,23 +235,18 @@ GDMonoAssembly *GDMonoAssembly::_load_assembly_from(const String &p_name, const
 }
 
 void GDMonoAssembly::_wrap_mono_assembly(MonoAssembly *assembly) {
-	String p_name = mono_assembly_name_get_name(mono_assembly_get_name(assembly));
-	GDMonoAssembly **existingassembly = GDMono::get_singleton()->get_loaded_assembly(p_name);
-
-	if (no_search || existingassembly != NULL) {
-		// Not sure whether the existingassembly check matters
-		// since no_search handles it in most cases?
-		// Can't hurt.
-		return;
-	}
+	String name = mono_assembly_name_get_name(mono_assembly_get_name(assembly));
 
 	MonoImage *image = mono_assembly_get_image(assembly);
-	mono_image_addref(image);
 
-	GDMonoAssembly *gdassembly = memnew(GDMonoAssembly(p_name, mono_image_get_filename(image)));
-	gdassembly->assembly = assembly;
-	gdassembly->loaded = true;
-	gdassembly->image = image;
+	GDMonoAssembly *gdassembly = memnew(GDMonoAssembly(name, mono_image_get_filename(image)));
+	Error err = gdassembly->wrapper_for_image(image);
+
+	if (err != OK) {
+		memdelete(gdassembly);
+		ERR_FAIL_V(NULL);
+	}
+
 	MonoDomain *domain = mono_domain_get();
```

Comment:
This is just some cleanup. I noticed we could call `wrapper_for_image` to avoid duplicated code. The `existingassembly` check is not needed. Moved the `no_search` condition to the hook function, to be consistent with the other hooks.